packages/clarity/test/nativeClarityBinProvider.ts

```diff
@@ -0,0 +1,47 @@
+import { assert } from "chai";
+import fs from "fs-extra";
+import { NativeClarityBinProvider, Receipt } from "../src";
+import { getTempFilePath } from "../src/utils/fsUtil";
+
+describe("NativeClarityBinProvider", () => {
+  it("create ephemeral", async () => {
+    const provider = await NativeClarityBinProvider.createEphemeral(
+      "node_modules/@blockstack/clarity-native-bin/.native-bin/clarity-sdk-v0.1.1/clarity-cli"
+    );
+    assert.isDefined(provider);
+  });
+
+  it("set initial allocations", async () => {
+    const principal = "ST12EY99GS4YKP0CP2CFW6SEPWQ2CGVRWK5GHKDRV";
+    const amount = 1000;
+    const tempDbPath = getTempFilePath("blockstack-local-{uniqueID}.db");
+    const provider = await NativeClarityBinProvider.create(
+      [{ principal, amount }],
+      tempDbPath,
+      "node_modules/@blockstack/clarity-native-bin/.native-bin/clarity-sdk-v0.1.0/clarity-cli"
+    );
+
+    const receipt = await getStxBalance(principal, provider);
+    assert.equal(receipt.result, `u${amount.toString()}`);
+  });
+});
+
+async function getStxBalance(
```

Comment:
Is there a better way to check the balance than creating an empty contract?

Reply:
I think the `eval_raw` command would need modified to support passing in a db file

Reply:
the documentation says that no db context is used for `eval_raw`. Is the empty contract a good solution?

Reply:
The rust code for `eval_raw` would ideally be updated to support a db context. This works for now though